cmd/mt-split-metrics-by-ttl/main.go

```diff
@@ -0,0 +1,101 @@
+	cassandraHostVerification = flag.Bool("cassandra-host-verification", true, "host (hostname and server cert) verification when using SSL")
+
+	cassandraAuth     = flag.Bool("cassandra-auth", false, "enable cassandra authentication")
+	cassandraUsername = flag.String("cassandra-username", "cassandra", "username for authentication")
+	cassandraPassword = flag.String("cassandra-password", "cassandra", "password for authentication")
+)
+
+func main() {
+	flag.Usage = func() {
+		fmt.Fprintln(os.Stderr, "mt-create-metrics-by-ttl [flags] ttl [ttl...]")
+		fmt.Fprintln(os.Stderr)
+		fmt.Fprintln(os.Stderr, "Creates schema of metric tables split by TTLs and")
+		fmt.Fprintln(os.Stderr, "assists in migrating the data to new tables.")
+		fmt.Println("Flags:")
+		flag.PrintDefaults()
+		os.Exit(-1)
+	}
+	flag.Parse()
+
+	var ttls []uint32
+	if flag.NArg() < 1 {
+		flag.Usage()
+	}
+
+	for i := 0; i < flag.NArg(); i++ {
+		ttl, err := strconv.Atoi(flag.Arg(i))
+		if err != nil {
+			flag.Usage()
+		}
+		ttls = append(ttls, uint32(ttl))
+	}
+
+	tmpDir, err := ioutil.TempDir(os.TempDir(), *cassandraKeyspace)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to get temp dir: %s", tmpDir))
+	}
+	snapshotDir := fmt.Sprintf("%s/snapshot", tmpDir)
+	err = os.Mkdir(snapshotDir, 0700)
+	if err != nil {
+		panic(fmt.Sprintf("Error creating directory: %s", err))
+	}
+
+	store, err := mdata.NewCassandraStore(*cassandraAddrs, *cassandraKeyspace, *cassandraConsistency, *cassandraCaPath, *cassandraUsername, *cassandraPassword, *cassandraHostSelectionPolicy, *cassandraTimeout, *cassandraReadConcurrency, *cassandraReadConcurrency, *cassandraReadQueueSize, 0, *cassandraRetries, *cqlProtocolVersion, 20, *cassandraSSL, *cassandraAuth, *cassandraHostVerification, ttls)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to instantiate cassandra: %s", err))
+	}
+	tables := store.GetTableNames()
+
+	// create directory/link structure that we need to define the future table names
+	err = os.Mkdir(fmt.Sprintf("%s/%s", tmpDir, *cassandraKeyspace), 0700)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to create directory: %s", err))
+	}
+	namedTableLinks := make([]string, 0)
+	for _, table := range tables {
+		namedTableLink := fmt.Sprintf("%s/%s/%s", tmpDir, *cassandraKeyspace, table)
+		namedTableLinks = append(namedTableLinks, namedTableLink)
+		err := os.Symlink(snapshotDir, namedTableLink)
+		if err != nil {
+			panic(fmt.Sprintf("Error when creating symlink: %s", err))
+		}
+	}
+
+	fmt.Println(fmt.Sprintf("The following tables have been created: %s", strings.Join(tables, ", ")))
+	fmt.Println("Now continue with:")
+	fmt.Println("- Stop metrictanks to stop writes")
+	fmt.Println(fmt.Sprintf("- Use `nodetool snapshot --table metric %s` on all cluster nodes to create snapshot of metric table", *cassandraKeyspace))
+	fmt.Println("  https://docs.datastax.com/en/cassandra/3.0/cassandra/tools/toolsSnapShot.html")
+	fmt.Println(fmt.Sprintf("- Move snapshot files to directory %s", snapshotDir))
```

Comment:
I updated it. Actually not everything the tool does needs to be run on each node, because the table schema only needs to be created once. But the message is now better at explaining what needs to be done on each node.